### ISSUE TITLE
Mistake in wilsonScore

### ISSUE THREAD

**Issue by evilive3000** [clux/decay#6]
In math notation there is a part `z^2 / 4n` you've already calculated it and cached into the variable `zzfn`. But then under the [square root](https://github.com/clux/decay/blob/master/decay.js#L58) you're dividing it again by `n`. 
So you've got something like `(z^2 / 4n) / n` finaly;

**Comment by clux** [comment]
Did a quick recheck and it does indeed look wrong. Damnit. I remember feeling so clever when I factored out that thing and thought it all balanced out.

Will change it for a more readable version, rather than try to be clever.

Thanks :disappointed:

**Comment by evilive3000** [comment]
It is readable. Actually I did the same optimization and were surprised when found you did the same ^_^. But results were slightly different so I started to dig.

**Comment by clux** [comment]
Hehe. Well I appreciate it at least.

I've added tests now that has been double checked with the python version from reddit.

In the end, I just rewrote it to follow the formula directly:

```js
    var sqrtexpr = ( p*(1-p) + z*z/(4*n) ) / n;
    return ( p + z*z/(2*n) - z*Math.sqrt(sqrtexpr) ) / (1+z*z/n);
```

But the mistake was dividing by `n` in the full sqrt expression and not just the `p*(1-p)` part inside the sqrt.

```js
# old wrong
(p + 2*zzfn - z*Math.sqrt((zzfn / n + p*(1 - p))/n)) / (1 + 4*zzfn);
# correct
(p + 2*zzfn - z*Math.sqrt(zzfn / n + p*(1 - p)/n)) / (1 + 4*zzfn);
```

I'll stick to the one mapping more closely onto the formula for now though.

Thanks again for the report :relieved: 

For anyone else wondering how big the change is, here was a printout of the failing tests before the change:

```
ok 7 1 ups 0 downs
not ok 8 10 ups 10 downs
  ---
    operator: eq
    expected: 0.2992949144298199
    actual: 0.31529391318907424
    at: null (/home/clux/repos/decay/test/order.test.js:29:5)
  ...
not ok 9 10 ups 0 downs
  ---
    operator: eq
    expected: 0.7224598312333834
    actual: 0.81734696184219
    at: null (/home/clux/repos/decay/test/order.test.js:30:5)
  ...
not ok 10 0 ups 10 downs
  ---
    operator: eq
    expected: 0
    actual: 0.09488713060880662
    at: null (/home/clux/repos/decay/test/order.test.js:31:5)
  ...
not ok 11 100 ups 50 downs
  ---
    operator: eq
    expected: 0.5878960768592671
    actual: 0.5889411445353994
    at: null (/home/clux/repos/decay/test/order.test.js:32:5)
  ...
not ok 12 50 ups 100 downs
  ---
    operator: eq
    expected: 0.2628864565745068
    actual: 0.2639315242506389
    at: null (/home/clux/repos/decay/test/order.test.js:33:5)
  ...
```

So yeah, not big, but it was wrong.

**Comment by clux** [comment]
Published in decay@1.0.12
